miden-lib/asm/sat/kernel.masm

```diff
@@ -110,82 +189,111 @@ end
 #! Panics if the ASSET is a fungible asset.
 #!
 #! Stack: [ASSET]
-#! Output: [has_asset]
+#! Output: [has_asset, 0, 0, 0]
 #!
 #! - ASSET is the non-fungible asset of interest
 #! - has_asset is a boolean indicating whether the account vault has the asset of interest
 export.account_vault_has_non_fungible_asset
+    push.0 movdn.4 push.0 movdn.4 push.0 movdn.4
+    # => [ASSET, 0, 0, 0]
+
     exec.account_vault::has_non_fungible_asset
+    # => [has_asset, 0, 0, 0]
 end
 
 #! Writes the assets of the currently executing note into memory starting at the specified address.
 #!
-#! Inputs: [dest_ptr]
+#! Inputs: [dest_ptr, 0]
 #! Outputs: [num_assets, dest_ptr]
 #!
 #! - dest_ptr is the memory address to write the assets.
 #! - num_assets is the number of assets in the currently executing note.
 export.get_note_assets
     exec.note::get_assets
+    # => [num_assets, dest_ptr, 0]
+
+    movup.2 drop
+    # => [num_assets, dest_ptr]
 end
 
 #! Returns the sender of the note currently being processed. Panics if a note is not being
 #! processed.
 #!
-#! Inputs: []
+#! Inputs: [0]
 #! Outputs: [sender]
 #!
 #! - sender is the sender of the note currently being processed.
 export.get_note_sender
     exec.note::get_sender
+    # => [sender, 0]
+
+    swap drop
+    # => [sender]
+
 end
```

Comment:
Similar to the above comment: I was thinking that this procedure would be available only in note context - but maybe there is no harm in making it available in account context as well.

Reply:
See response to `get_note_assets` comment.  I think we should decide on a access standard and apply consistently.